Pylint: Inconsistent return types in `spice_exception_handler` decorator.

## 💡 Description
Fix the pylint issue `inconsistent-return-statements`. The run `pylint . --disable=all --enable=inconsistent-return-statements` reports:
```
pds/naif_pds4_bundler/utils/decorators.py:20:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
```
This is caused by the handling of the SpiceyPyError exception, which does not clearly indicate that a new exceptions is raised.

--- a/src/pds/naif_pds4_bundler/pipeline/runtime.py
+++ b/src/pds/naif_pds4_bundler/pipeline/runtime.py
@@ -2,7 +2,7 @@ from __future__ import annotations
 import logging
 import os
 from pathlib import Path
-from typing import Optional, TYPE_CHECKING
+from typing import NoReturn, Optional, TYPE_CHECKING
 
 import spiceypy
 
@@ -108,7 +108,7 @@ def finish_execution(setup: Setup, log_manager: Log) -> None:
     log_manager.stop()
 
 
-def handle_npb_error(message: str, setup: Optional[Setup] = None) -> None:
+def handle_npb_error(message: str, setup: Optional[Setup] = None) -> NoReturn:
     """Signal a NPB error and write run artifacts.
 
     Side effects:

